addons/cargo/functions/fnc_onMenuOpen.sqf

```diff
@@ -22,37 +22,64 @@ params ["_display"];
 uiNamespace setVariable [QGVAR(menuDisplay), _display];
 
 if (GVAR(interactionParadrop)) then {
-    (_display displayCtrl 12) ctrlSetText (localize LSTRING(paradropButton));
+    (_display displayCtrl 12) ctrlSetText LLSTRING(paradropButton);
 };
 
 [{
+    params ["_vehicle", "_pfhID"];
+
     disableSerialization;
+
     private _display = uiNamespace getVariable QGVAR(menuDisplay);
-    if (isnil "_display") exitWith {
-        [_this select 1] call CBA_fnc_removePerFrameHandler;
+
+    if (isNil "_display") exitWith {
+        _pfhID call CBA_fnc_removePerFrameHandler;
     };
 
-    if (isNull GVAR(interactionVehicle) || {(([ACE_player, GVAR(interactionVehicle)] call EFUNC(interaction,getInteractionDistance)) >= MAX_LOAD_DISTANCE) && {(vehicle ACE_player) != GVAR(interactionVehicle)}}) exitWith {
+    // Close menu if in invalid state
+    if (
+        !alive _vehicle ||
+        {locked _vehicle >= 2} ||
+        {!(_vehicle getVariable [QGVAR(hasCargo), getNumber (configOf _vehicle >> QGVAR(hasCargo)) == 1])} ||
```

Comment:
Possible to check against `vehicle getVariable [QGVAR(loaded), []]` instead here to avoid the config lookup? Though I don't see the need to close the menu if Cargo is empty.

Reply:
> Possible to check against `vehicle getVariable [QGVAR(loaded), []]` instead here to avoid the config lookup? Though I don't see the need to close the menu if Cargo is empty.

I don't think it's possible to use `QGVAR(loaded)` instead, as they don't serve quite the same purpose. It would mean you can't open the cargo menu if there are no items loaded.

We could cache all of the config entries in hashmaps.

Reply:
Remind me what `QGVAR(hasCargo)` does at all then, because it seems redundant

Reply:
It allows the cargo of a vehicle to be enabled/disabled.

Reply:
Set that variable during initVehicle/initObject instead then?

Reply:
https://github.com/acemod/ACE3/pull/9617/files#diff-ebaa31fadbcbd31a401d08ae98225fd53bc7a290710ce2cd6899e4311be561b4R41 worries me a bit - I'm honestly not quite sure if the comments means that the variable is API or not. Either way, it can be changed mid mission using `FUNC(setSpace)`.

Reply:
Not public. Should only be touched by `FUNC(setSpace)` (which if set to 0 disables cargo and thus should disable the menu). Go wild.

Reply:
It no longer checks for the config entry, as it assumes the condition to open the menu already did.